support/vm/vm.go

```diff
@@ -340,8 +343,23 @@ type MessageResult struct {
 	GasCharged int64
 }
 
-// ApplyMessage applies the message to the current state.
-func (vm *VM) ApplyMessage(from, to address.Address, value abi.TokenAmount, method abi.MethodNum, params interface{}) MessageResult {
+func (vm *VM) ApplyMessage(from, to address.Address, value abi.TokenAmount, method abi.MethodNum, params interface{}, info string) (MessageResult, error) {
```

Comment:
Returning errors instead of panicing seems like the way to go

I can add a testing helper `RequireApplyMessage(t, v, ...)` to make this much less annoying